### PR TITLE
feat: update operator workflows

### PR DESCRIPTION
**Pull request by yanksyoon**
This PR updates the actions workflows to match the latest versions of the operator-workflows.

### CODE REVIEW

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by arturo-seijas — DISMISSED** [review]

File: .github/workflows/publish_charm.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Publish to edge
+
+# On push to a "special" branch, we:
+# * always publish to charmhub at latest/edge/branchname
+# * always run tests
```
Comment (gregory-schiano) [inline .github/workflows/publish_charm.yaml:5]: Does it still run the tests ?

File: .github/workflows/publish_charm.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Publish to edge
+
+# On push to a "special" branch, we:
+# * always publish to charmhub at latest/edge/branchname
```
Comment (gregory-schiano) [inline .github/workflows/publish_charm.yaml:4]: It pushes to "latest/edge" not in "latest/edge/branchname" as far as I see

File: .github/workflows/publish_charm.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# * always run tests
+# where a "special" branch is one of main/master or track/**, as
```
Comment (gregory-schiano) [inline .github/workflows/publish_charm.yaml:6]: Only `main` is targeted here

Comment (yanksyoon) [reply .github/workflows/publish_charm.yaml]: Updated the comments, thank you! I'll issue a PR against the charms requiring the comment changes as well :)

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by weiiwang01 — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]

**Pull request merged**